### DISCUSSION TITLE
Why `RepeatDataset` can reduce time of loading data between epochs

### DISCUSSION THREAD

**Discussion by MciaR** [open-mmlab/mmengine#1193] [Q&A] [answered]
I find this note when using `RepeatDataset`, which is in `mmengine/dataset/dataset_wrapper.py`.
```python
class RepeatDataset:
    """A wrapper of repeated dataset.

    The length of repeated dataset will be `times` larger than the original
    dataset. This is useful when the data loading time is long but the dataset
    is small. Using RepeatDataset can reduce the data loading time between
    epochs.
    """
```

When a epoch is done, the sampler of Dataloader will just be reset in my opion, so it's seem to take no extra time to reload new data. Why the note says it is helpful to reduce the data loading time between epochs. 
Thanks for comments!

**Comment by HAOCHENYE** [comment, marked as answer]
Sorry for my late reply. If the `persistent_workers` is set to False in [`Dataloader`](https://github.com/open-mmlab/mmagic/blob/9d374535c1224e4607cee474e9fdff528559054e/mmagic/models/editors/singan/singan.py#L488), there could be additional overhead to relaunch the workers. However, if you enable it in `Dataloader`, it means all workers will run independently in multiple processes, and any modification to dataset or pipeline will only work in the main process, and the dataset and pipeline in subprocesses launched by Dataloader will not be influenced.

**Reply by MciaR** [reply]
Get it! Thanks for your answer!
